docs/application-connector/docs/08-04-register-secured-api.md

```diff
@@ -51,3 +53,46 @@ This is an example of the `api` section of the request body for an API secured w
             },
         }  
 ```
+
+## Register an API secured with generated client certificates
+
+To register an API and secure it with generated client certificates, you must add the `credentials.certificateGen` object to the `api` section of the service registration request body.
+You must include this field:
+
+| Field   |  Description |
+|:----------:|:------|
+| **commonName** |  Name of the generated certificate.  |
+
+This is an example of the `api` section of the request body for an API secured with generated client certificates:
+
+```
+    "api": {
+        "targetUrl": "https://sampleapi.targeturl/v1",
+        "credentials": {
+            "certificateGen": {
+                "commonName": "{CERT_NAME}"
+            },
+        }  
+```
+
+>**NOTE:** If you update the registered API and change the `certificateGen.commonName`, the Application Registry generates a new certificate-key pair for that API. When you delete an API secured with generated client certificates, the Application Registry deletes the corresponding certificate and key.
+
+### Details
+
+When you register an API with the `credentials.certificateGen` object, the Application Registry generates a SHA256withRSA-encrypted certificate and matching key. The certificate and key pair is stored in a Secret in the `kyma-integration` Namespace. To fetch the certificate and key encoded with base64, run this command:
+
+```
+kubectl -n kyma-integration get secrets app-{APP_NAME}-{SERVICE_ID}
+```
+
+>**NOTE:** Replace the `APP_NAME` placeholder with the name of the Application used to connect the external solution that is the origin of the API. Replace the `SERVICE_ID` placeholder with the ID of the registered service to which the API belongs. You get this ID after you register an external solution's service in Kyma.
+
+To enable communication between Kyma and an API secured with this authentication method, set the generated certificate as a valid authentication medium for all calls coming from Kyma in your external solution.
```

Comment:
Please highlight that we have now two possibilities:
- use generated client cert
- insert your own one

Reply:
.. you can use the generated certificate if your API allow inserting own certificates
... if you need to have a better certificate or your API provided you one, you can easyly update the secret and provide the certificate of your choice

Something like that will be cool.